### PR TITLE
Support both Prost and rust-protobuf

### PR DESCRIPTION
**Pull request by nrc**
Should probably block on https://github.com/pingcap/raft-rs/pull/262 landing to avoid the git dep to my branch. This will allow us to work on both Prost and rust-protobuf on master.

### CODE REVIEW

**Comment by ice1000** [comment]
How are we supposed to review this?

**Comment by nrc** [comment]
> How are we supposed to review this?

You can ignore the code in src/protobuf - it is all auto-generated. If you are very keen you can pull the branch, delete those files run `cargo build --features gen` and check they are the same.

The rest of the PR is not too huge.

File: build.rs [hunk 1]
```diff
@@ -68,6 +69,18 @@ fn main() {
...
+    generate_protobuf_files(&file_names, "src/protobuf");
+    let mod_names = module_names_for_dir("src/protobuf");
+
+    let out_file_names: Vec<_> = mod_names
+        .iter()
+        .map(|m| format!("src/protobuf/{}.rs", m))
+        .collect();
+    let out_file_names: Vec<_> = out_file_names.iter().map(|f| &**f).collect();
+    replace_read_unknown_fields(&out_file_names);
+    generate_lib_file(&mod_names);
```
Comment (ice1000) [inline build.rs]: and then ` protobuf_build::rustfmt(Path::new("src/protobuf.rs"));`?

**Comment by BusyJay** [comment]
Close as it's duplicated with #427.

**Pull request closed without merging**